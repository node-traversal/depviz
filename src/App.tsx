import React, { FunctionComponent } from 'react';
import logo from './logo.svg';
import './App.css';
import CodeTree from './CodeTree';
import data from './sample-data.json';
import toTree from "./toTree";

const treeOptions = {
    noTraverse: {
        '@material-ui/core/styles/useTheme': true,
        '@material-ui/core/styles/defaultTheme': true,
        '@material-ui/core/InputBase': true,
        '@material-ui/core/Modal': true,
        '@material-ui/core/styles/withStyles': true
    },
    hidden: {
        clsx: true,
        // react: true,
        // 'react-dom': true,
        'prop-types': true
    }
};

type AppProps = {
}

const App: FunctionComponent<AppProps> = () => {
    const tree = toTree(data, treeOptions);

    return (
        <div className="App">
          <header className="App-header">
            <img src={logo} className="App-logo" alt="logo" />
            <CodeTree dependencies={tree}/>
          </header>
        </div>
  );
};

export default App;
