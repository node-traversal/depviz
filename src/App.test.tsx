import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

it('renders application skeleton', () => {
  const wrapper = shallow(<App/>);
  expect(wrapper.debug()).toMatchSnapshot();
});
