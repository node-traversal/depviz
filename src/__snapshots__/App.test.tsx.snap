// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders application skeleton 1`] = `
"<div className=\\"App\\">
  <header className=\\"App-header\\">
    <img src=\\"A MOCK SVG\\" className=\\"App-logo\\" alt=\\"logo\\" />
    <CodeTree dependencies={{...}} />
  </header>
</div>"
`;
